tests/components/fitbit/test_sensor.py

```diff
@@ -532,6 +533,27 @@ async def test_nutrition_scope_config_entry(
     assert state
     assert (state.state, state.attributes) == snapshot
 
+@pytest.mark.parametrize(
+    ("scopes", "unit_system"),
+    [(["nutrition"], METRIC_SYSTEM), (["nutrition"], US_CUSTOMARY_SYSTEM)],
+)
+async def test_nutrition_scope_config_entry_missing_key(
+    hass: HomeAssistant,
+    setup_credentials: None,
+    integration_setup: Callable[[], Awaitable[bool]],
+    unit_system: UnitSystem,
+) -> None:
+    """API to return unexpected key so integration raises KeyError."""
+    hass.config.units = unit_system
+
+    with (
```

Comment:
This doesn't seem right yet given ti raises an exception nd the point of the pr is to not raise this exception